docs/developer/decision-records/2022-06-13-registration-service/README.md

```diff
@@ -0,0 +1,137 @@
+
+Responsibilities of the Dataspace Authority in MVD version:
+- Establishes a DID defining the Dataspace.
+- Manages policies defining who is allowed to participate in the dataspace.
+- Manages enrollment process of the Dataspace participants.
+- Lists the available Dataspace participants.
+
+## Architecture 
+
+The MVD's Registration Service will be written in Java and re-use the runtime framework and modules from EDC. This enables us to leverage the same functionality such as policy validation, re-use domain objects and architectural principles.
+
+Potentially long-running operations such as onboarding and offboarding must be asynchronous and are handled using the state machine concept. The state 
+machine is operated in such a way that domain objects are loaded from storage, processed and then put back into storage to make the registration service runtime stateless.
+
+In the MVD version the Registration Service will use in-memory storage.
+
+## Deployment
+
+The Registration Service will be deployed as a separate component in the Dataspace environment.
+
+## Identity
+
+The Registration Service will have its own `did:web` document deployed, that points to the enrollment API endpoint.
+
+Please note that in the future versions the `did:web` document points to the Self-Description Document url that allows to retreive the enrollment API endpoint. 
+MVD version is using a simplification omitting the Self-Description Document in the process.
+
+## Operations
+
+This document specifies only operations supported by the Registration Service in MVD version:
+- Dataspace participant onboarding
+- Listing all Dataspace participants
+
+Other operations that will be supported in future versions of the Registration Service:
+- Dataspace participant offboarding
+- Blacklisting
+
+### 1. Onboarding
+
+#### Participants
+
+1. Company1, an entity which intends to become a Dataspace participant
+2. The Dataspace Authority, which manages the enrollment process
+
+#### Overview
+
+A Client for Company1 initiates the enrollment process by resolving and contacting the enrollment API endpoint for the Dataspace Authority. The client could be e.g. a CLI utility.
+
+The Dataspace Authority enrollment service obtains Verifiable Credentials from Company1 to determine whether it meets enrollment policies. The enrollment service then issues a Verifiable Credential that establishes membership and pushes it to Company 1's Identity Hub, and stores membership and certificate information.
+
+In simple scenarios, enrollment could be fast and fully automated. However, in advanced scenarios, enrollment policies could require interactions with external systems, and even manual processes. Therefore, it is implemented asynchronously.
+
+There could be different "types" of onboarding, e.g. onboarding a participant or a credential issuer, so the architecture has to support that.
+
+#### Flow sequence
+
+![dataspace-enrollment](dataspace-enrollment.png)
```

Comment:
the diagram shows a sync version of the flow. Update to account for async behavior? E.g. participant is stored in registry immediately, and 6 is actually a state update

"Participant registry" should probably be renamed to "Participant store" or similar